aws/get_running_instances.py

```diff
@@ -0,0 +1,88 @@
+"""
+Use boto to get the number of running or pending EC2 instances matching a key
+name.
+
+Usage:
+    If you've defined the AWS credentials as environment variables or in a
+    .boto file, then use:
+        `python get_running_instances.py -k some-key-name`
+
+    Else, you can add the aws keys as arguments to the above command:
+        ** '--aws-access-key-id' or '-i'
+        ** '--aws-secret-access-key' or '-s'
+"""
+import argparse
+import boto
+import logging
+import os
+import sys
+
+
+logging.basicConfig(format='[%(levelname)s] %(message)s')
+logger = logging.getLogger(__name__)
+
+
+def get_running_instances(key_name, aws_access_key_id, aws_secret_access_key):
+    """
+    Logs and returns (as a string) the number of EC2 instances matching
+    `key_name` that are running or pending.
+
+    :Args:
+        key_name: the ec2 key_name to match.
+        aws_access_key_id
+        aws_secret_access_key
+    """
+    ec2conn = boto.connect_ec2(aws_access_key_id, aws_secret_access_key)
+    reservations = ec2conn.get_all_instances()
```

Comment:
Naming convention: we should use something else rather than 'reservations'. That might confuse it with 'reserved instances' which are a specific kind of EC2 unit (one you buy ahead of time by paying for time in bulk).